patches/tModLoader/Terraria/GameContent/UI/GameTipsDisplay.cs.patch

```diff
@@ -0,0 +1,54 @@
+--- src/TerrariaNetCore/Terraria/GameContent/UI/GameTipsDisplay.cs
++++ src/tModLoader/Terraria/GameContent/UI/GameTipsDisplay.cs
+@@ -4,11 +_,12 @@
+ using System.Collections.Generic;
+ using Terraria.GameInput;
+ using Terraria.Localization;
++using Terraria.ModLoader;
+ using Terraria.UI.Chat;
+ 
+ namespace Terraria.GameContent.UI
+ {
+-	public class GameTipsDisplay
++	public partial class GameTipsDisplay
+ 	{
+ 		private class GameTip
+ 		{
+@@ -67,6 +_,7 @@
+ 			_tipsGamepad = Language.FindAll(Lang.CreateDialogFilter("LoadingTips_GamePad."));
+ 			_tipsKeyboard = Language.FindAll(Lang.CreateDialogFilter("LoadingTips_Keyboard."));
+ 			_lastTip = null;
++			Initialize();
+ 		}
+ 
+ 		public void Update() {
+@@ -137,6 +_,7 @@
+ 
+ 		private void AddNewTip(double currentTime) {
+ 			string textKey = "UI.Back";
++			/*
+ 			List<LocalizedText> list = new List<LocalizedText>();
+ 			list.AddRange(_tipsDefault);
+ 			if (PlayerInput.UsingGamepad)
+@@ -146,8 +_,20 @@
+ 
+ 			if (_lastTip != null)
+ 				list.Remove(_lastTip);
++			*/
++			List<GameTipData> list = new List<GameTipData>();
++			list.AddRange(allTips);
++			if (PlayerInput.UsingGamepad)
++				list.RemoveAll(tip => tip.Mod == "Terraria" && tip.TipText.Key.StartsWith("LoadingTips_Keyboard"));
++			else
++				list.RemoveAll(tip => tip.Mod == "Terraria" && tip.TipText.Key.StartsWith("LoadingTips_GamePad"));
```

Comment:
would be better if we had some sort of 'default visibility' system like player layers.